packages/opentelemetry-exporter-collector-grpc/README.md

```diff
@@ -18,12 +18,15 @@ npm install --save @opentelemetry/exporter-collector-grpc
 The CollectorTraceExporter in Node expects the URL to only be the hostname. It will not work with `/v1/trace`.
 
 ```js
+const Graceful = require('node-graceful');
```

Comment:
Is it really needed to refer to a 3rd party module in the readme? What happens if a user is not using this?

Reply:
FWIW I've been looking into various 3rd party libs for this sort of thing, and am settling without a lib with the following which is clean enough for me:

```js
[
  'exit',
  'SIGINT',
  'SIGUSR1',
  'SIGUSR2',
  'SIGTERM',
  'SIGHUP',
  'uncaughtException',
].forEach(event => {
  process.on(event, () => provider.shutdown().catch(console.error);
});
```

Reply:
that's going to be very verbose for example/README code :/ -- any suggestions on how to handle?

Reply:
alternatively, we could expose a convenience method on provider to hook. thoughts?

Reply:
Does an example really need signal handling?

Reply:
I feel like it does, especially if we just make it easier to register the signal handlers as a method on provider, rather than needing a bunch of code.

otherwise people will lose spans on shutdown, and not know why. better to make the example code people are likely to cut-paste do the right thing out of box.

Reply:
> that's going to be very verbose for example/README code :/

edited my snippet to handle SIGINT and SIGTERM only which should be good enough so it's less verbose now, good enough for README imo

Reply:
I agree that those 2 signals are fine. It avoids "recommending" a dependency and is more clear why it's there for curious readers.

Reply:
perfect, I'll change out the library for the hook.

Reply:
I think for a README in an exporter package it's even enough to document that shutdown/flush/.. needs to be called to avoid data loss. A lot application have already their own shutdown mechanism in place where they just have to add OTel components.

Besides that shutting down TracerProvider directly is maybe not the right thing. Users of `opentelemetry-sdk-node` should actually use `sdk.shutdown()` instead.

Reply:
yes. but for the example of a BasicTracerProvider that's been instantiated outside an SDK, it should be shut down.